test/performance/performance.go

```diff
@@ -0,0 +1,333 @@
+			fmt.Sprintf("%.5f", entry.apiserverAvg),
+			fmt.Sprintf("%.5f", entry.apiserverMax),
+		}
+		if err := w.Write(line); err != nil {
+			klog.Exitf("Error writing data to file; %s", err)
+		}
+	}
+}
+
+func init() {
+	klog.SetOutput(os.Stdout)
+	klog.InitFlags(flag.CommandLine)
+}
+
+func main() {
+	// pull in test variables from flags
+	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)
+
+	var plc_filename, output_filename string
+	var n_per_batch, n_total, per_batch_sleep int
+
+	pflag.StringVarP(&plc_filename, "policy", "p",
+		performance_dir+"/resources/templates/deployment-plc.yaml", "path to test policy YAML, scoped to performance directory")
+	pflag.IntVarP(&n_per_batch, "policies-per-batch", "b", 100, "number of policies to create per batch")
+	pflag.IntVarP(&n_total, "total-policies", "t", 1000, "total number of policies created")
+	pflag.IntVarP(&per_batch_sleep, "sleep", "s", 20, "time (min) to sleep after creating a batch of policies")
+	pflag.StringVar(&output_filename, "csv", performance_dir+"results.csv", "path to CSV to export results to")
+
+	pflag.Parse()
+
+	// wait for config map to be applied properly before executing command
+	_, err := bash("oc get secret -n openshift-user-workload-monitoring | grep  prometheus-user-workload-token")
+	if err != nil {
+		klog.V(2).Info("Setting up secret for k8s metrics")
+		_, err = exec.Command("kubectl", "apply", "-f", performance_dir+"/resources/setup/metrics-configmap.yaml").CombinedOutput()
+		if err != nil {
+			klog.Exitf("Error applying metrics configMap: %s", err)
+		}
+		// wait for secret to be created for 2 minutes
+		found := false
+		for i := 0; i < 120; i += 5 {
+			_, err = bash("oc get secret -n openshift-user-workload-monitoring | grep  prometheus-user-workload-token")
+			if err == nil {
+				found = true
+				break
+			}
+			klog.V(2).Info("Secret not found, waiting 5 seconds...")
+			time.Sleep(5 * time.Second)
+		}
+
+		if !found {
+			klog.Fatal("Error: Prometheus secret creation timed out")
+		}
+	}
+
+	secret, err := bash("oc get secret -n openshift-user-workload-monitoring | grep  prometheus-user-workload-token | head -n 1 | awk '{print $1 }'")
+	if err != nil {
+		klog.Exitf("Error getting Prometheus metrics secret: %s", err)
+	}
+
+	token, err := bash(fmt.Sprintf("echo $(oc get secret %s -n openshift-user-workload-monitoring -o json | jq -r '.data.token') | base64 -d", secret))
+	if err != nil {
+		klog.Exitf("Error getting API token from secret: %s", err)
+	}
+
+	thanos_host, err := bash("oc get route thanos-querier -n openshift-monitoring -o json | jq -r '.spec.host'")
+	if err != nil {
+		klog.Exitf("Error getting Thanos host: %s", err)
+	}
+
+	klog.Info("Setup complete! Starting the Config Policy Controller performance test :)")
+	table_data := []metricData{}
+
+	table_data = append(table_data, getMetrics(thanos_host, token, per_batch_sleep, 0))
+
+	for total_plcs := 0; total_plcs < n_total; total_plcs += n_per_batch {
```

Comment:
What do you think about incrementing `total_plcs` after each policy is created rather than the `for` loop performing it?

The line at the end of the loop that included `total_plcs+n_per_batch` initially confused me.  :smile: